kernel_tuner/util.py

```diff
@@ -284,9 +284,15 @@ def check_restrictions(restrictions, params: dict, verbose: bool) -> bool:
                 # look up the selected parameters and their value
                 selected_params = dict((key, params[key]) for key in selected_params)
                 # call the restriction
-                if not restrict(**selected_params):
-                    valid = False
-                    break
+                if isinstance(restrict, Constraint):
+                    restrict = convert_constraint_restriction(restrict)
+                    if not restrict(list(selected_params.values())):
+                        valid = False
+                        break
+                else:
+                    if not restrict(**selected_params):
+                        valid = False
+                        break
```

Comment:
It seems our automated linting system 'SonarCloud' isn't too happy about this level of nesting control flow. Maybe we can break things up into smaller chunks?

Reply:
I've refactored this into two separate functions, SonarCloud is happy again.